### PR TITLE
Create dummy logs for all test cases

### PR DESCRIPTION
**Pull request by xiulipan**
- try to get error trace for all test case
- create dummy files after log init

### CODE REVIEW

**Review by marc-hb — COMMENTED** [review]
As this PR does not call the new function from anywhere yet I'm not sure where this is headed.

I like the idea of always creating log files to make log collection logic much simpler, however I think the creation of each log file should be in a single place, in other words I think `dmesg.txt` should only and always be created by `hijack.sh#func_exit_handler()`  and `slogger.txt` + `etrace.txt` should only and always be created at the start of `lib.sh#func_lib_start_log_collect()`. This makes the rest of the program logic simpler because if these functions have already been called then the files exist, if they have not been called yet then the log files do not exist yet. Making sure that these files have been created is also simpler: no need to look at logic spread across multiple places.

File: case-lib/logging_ctl.sh [hunk 1]
```diff
@@ -63,6 +63,22 @@ _func_log_directory()
...
+    # do not create dummy files if three is no LOG_ROOT
+    if [ -n "$LOG_ROOT" ]; then
+        # create dummy files
+        for filename in "${LOG_LIST[@]}"
+        do
+            echo "dummy file $filename" > "$LOG_ROOT/$filename"
+        done
+    else
+        return
+    fi
```
[the remark this answers is not in the document]
Comment (xiulipan) [reply case-lib/logging_ctl.sh:79]: I will refine the code here.

File: case-lib/logging_ctl.sh [hunk 2]
```diff
@@ -63,6 +63,22 @@ _func_log_directory()
...
+}
+
 for _func_ in $(declare -F|grep _func_log_|awk '{print $NF;}')
```
Comment (xiulipan) [inline case-lib/logging_ctl.sh:82]: check this for loop. It call all `_func_log_` functions when we source this file.

**Comment by xiulipan** [comment]
@marc-hb II remember your request about the dummy files is to create dummy files for all possible existing logs so we can report missing files easily in our framework. Or you have some new idea here?

PS: please check our code and not all test case call `lib.sh#func_lib_start_log_collect()`

For caller question, please see my inline comment

**Comment by marc-hb** [comment]
> for _func_ in $(declare -F|grep _func_log_|awk '{print $NF;}')

Wow! No, I did not read the whole file and I did not notice this, thanks. It is some impressive way to defeat `git grep` and obfuscate code. I just ran it and verified that this for loop runs only 2  functions! Please get rid of it and replace with plain calls if you get a chance.

> PS: not all test case call lib.sh#func_lib_start_log_collect()

This is the much more important issue that should be fixed first. Dummy log files are useful to simplify log collection but the much important goal is to collect actual logs for every single test by default (unless explicitly disabled on the command line)

**Comment by xiulipan** [comment]
The real problem here is that logger collection is not capable for all test case:

- we have **verify-logger** test. Which means we may not able to run the logger test before this test case pass
- kmod test will hang if we are collection logger trace at background

So we make the logger collection as an optional setting in our test cases.

**Pull request closed without merging**